### PR TITLE
TEP 0003: Tekton Catalog Organization

### PR DESCRIPTION
**Pull request by vdemeester**
/cc @sthaha @bobcatfish 

Depends on #121 to be merged (also #122)

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

### CODE REVIEW

File: teps/0003-tekton-catalog-organization.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+
+Guidelines aim to help users author their tekton resource. Requirements are *applying* those requirements to the upstream catalog. The requirements are not part of the Catalog contract but they should be applied as much as possible to other catalogs.
+
+Automation might be run against all verified and official resources to ensure (see [Tekton Catalog Test Infrastructure](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)):
+
+1. Resources have all description fields filled in
+
+2. Resources must have app.kubernetes.io /version to indicate the version of the resource
+
+3. Resources can include annotation.tags which is a comma separated tags associated with it e.g. kaniko task can have a tag: container-image, build-tool
```
Comment (bobcatfish) [inline teps/0003-tekton-catalog-organization.md:315]: can we make these `tekton.dev/tags`? 

https://github.com/tektoncd/catalog/pull/389 mentions `displayName` as well, same suggestion there

+ @PuneetPunamiya

Comment (sthaha) [reply teps/0003-tekton-catalog-organization.md:315]: These are 2 minor changes that we can incorporate, before #tekoncd/catalog reorg merges - https://github.com/tektoncd/catalog/pull/389 . Hub preview will need some work but I think we should use tekton specific annotations.

Comment (vdemeester) [reply teps/0003-tekton-catalog-organization.md]: Updated

**Pull request merged**